addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -1001,7 +1001,7 @@ export const formatSelection = (editor, formatName, {applyStyle, formatProps} =
     }
 
     const selectedTextNodes = getSelectedNodes(editor.editable)
-        .filter(n => n.nodeType === Node.TEXT_NODE && closestElement(n).isContentEditable);
+        .filter(n => n.nodeType === Node.TEXT_NODE && closestElement(n).isContentEditable && isVisibleTextNode(n));
```

Comment:
Zero-width spaces are not considered visible anymore so this is likely the problem. I'll test it.